src/main/kotlin/com/exactpro/th2/codec/xml/XmlPipelineCodec.kt

```diff
@@ -371,6 +364,32 @@ open class XmlPipelineCodec : IPipelineCodec {
         nodes: List<Node>,
         fieldStructure: IFieldStructure
     ) {
+        if (fieldStructure is IMessageStructure && fieldStructure.isVirtual()) {
+            val fields = HashMap<String, Queue<Node>>()
+            nodes.forEach {
+                fields.computeIfAbsent(it.nodeName) { LinkedList() }.add(it)
+            }
+
+            if (fieldStructure.isCollection) {
+                val listValue = ArrayList<Message.Builder>()
+                while (fields.size > 0) {
+                    val virtualMessage = message()
+                    decodeVirtualMessage(fields, virtualMessage, fieldStructure)
+                    listValue.add(virtualMessage)
+                }
+
+                message[fieldName] = listValue
+            } else {
+                val virtualMessage = message()
+                decodeVirtualMessage(fields, virtualMessage, fieldStructure)
+                if (fields.isNotEmpty()) {
+                    throw DecodeException("Can not decode field with name '$fieldName'. Not all field decoded for virtual message with name '${fieldStructure.name}'")
+                }
+                message[fieldName] = virtualMessage
```

Comment:
Let's implement the logic
1) create a message
2) check does the current message contain the current node
2-yes.1) check if the current message structure collection
2-yes.1-yes) store current message, create new, go to 2-no.1
2-yes.1-no) throw the exception "... the node has duplicated field ..."
2-no.1) put field, go to the next initiation